### ISSUE TITLE
[pixeldata] Enable building for WebAssembly

### ISSUE THREAD

**Issue by Enet4** [Enet4/dicom-rs#130]
This issue also depends on #121, but gating GDCM might be insufficient to ensure that it builds on WebAssembly. As `rayon` is used for parallel processing of pixel data, it may attempt to spawn threads, which [might be incompatible with this environment](https://rustwasm.github.io/book/reference/which-crates-work-with-wasm.html#spawning-threads).

**Comment by pevers** [comment]
> This issue also depends on #121, but gating GDCM might be insufficient to ensure that it builds on WebAssembly. As `rayon` is used for parallel processing of pixel data, it may attempt to spawn threads, which [might be incompatible with this environment](https://rustwasm.github.io/book/reference/which-crates-work-with-wasm.html#spawning-threads).

Yes I think you are correct. Maybe we can configure the threadpool for rayon to just use the main thread.

**Comment by Enet4** [comment]
> Yes I think you are correct. Maybe we can configure the threadpool for rayon to just use the main thread if we compile for wasm.

Ah, that too might work. I was picturing a feature gate on all of `rayon` and replace the algorithms to use plain iterators, but that would be less invasive.

**Comment by pevers** [comment]
I found this crate: https://github.com/GoogleChromeLabs/wasm-bindgen-rayon

And it works!

![Screenshot 2021-05-20 at 21 46 50](https://user-images.githubusercontent.com/3428649/119040202-659ded80-b9b5-11eb-9e7c-d5e1a001ece4.png)

I did a test and managed to compile to WASM by making GDCM optional. My fork is here: https://github.com/pevers/dicom-rs/tree/pevers/gdcm-feature . I'll open a PR soon. It is work in progress.

And this is the test code: https://github.com/pevers/dicom-wasm

There is quite a list of caveats, but it doesn't require to touch the dicom-rs rayon implementation. And it should support multi-threading by webworkers 🎉 .
